### PR TITLE
Test scrollbar is ignored in width constraints (#762)

### PR DESCRIPTION
**Pull request by ctjhoa**
https://github.com/Addepar/ember-table/issues/762
This is a draft PR which doesn't include tests.
It's mostly to start discussion around the solution and the global design.

### CODE REVIEW

File: addon/-private/column-tree.js [hunk 1]
```diff
@@ -630,8 +630,10 @@ export default EmberObject.extend({
     }
 
     let containerWidthAdjustment = get(this, 'containerWidthAdjustment') || 0;
+    let containerScrollbarWidth = getScrollbarWidth(this.container);
+    let containerInnerWidth = getInnerClientRect(this.container).width;
     let containerWidth =
-      getInnerClientRect(this.container).width * this.scale + containerWidthAdjustment;
+      (containerInnerWidth - containerScrollbarWidth) * Number(Math.round(this.scale+'e2')+'e-2') + containerWidthAdjustment;
```
Comment (ctjhoa) [inline addon/-private/column-tree.js]: I do not fully understand the purpose of `this.scale`.
The problem is that the `getScale` function compare `container.offsetHeight` and `container.getBoundingClientRect().height`, depending on the browser/window size, those values can be computed as integers or float.
So in some cases, you could have a float value for `container.getBoundingClientRect().height` and an integer value for `container.offsetHeight` which result on a scale not accurate.

2019-09-11-142146_1364x237_scrot

File: addon/components/ember-tbody/component.js [hunk 2]
```diff
@@ -260,6 +261,12 @@ export default Component.extend({
     );
   },
 
+  _updateColumnTree: observer('wrappedRows.[]', function() {
+    next(() => {
```
Comment (ctjhoa) [inline addon/components/ember-tbody/component.js]: This timer is flaky. We need to wait for all table rows to be rendered but sometimes rows aren't rendered yet.

Comment (ctjhoa) [reply addon/components/ember-tbody/component.js]: Using `scheduler.schedule('affect'` & `Ember.run.next` is the only solution found to wait for rendered rows.

**Comment by ctjhoa** [comment]
This has been manually tested on Edge, firefox, chrome and Safari.

**Comment by ctjhoa** [comment]
@mixonic
Hi, I've had some times to work on this.
So I have now a better understanding of the sizes computations but correct me if I'm wrong.
Browsers aren't consistent in their way to expose sizes of an element. For example, they expose `offsetWidth` as integer and `getBoundingClientRect().width` as float. Unfortunately, there isn't always an integer/float equivalent exposed.
`scale` seems to be a ratio which allows to go between integer and float values.

My first version had wrong assumptions on those size types and that's why I needed to round the scale to get accurate result. This should be fixed now.

I've added tests too to demonstrate the initial issue. However, I didn't change the observer yet. This will be my future task. Anyway it was just to give you updated status.

**Comment by ctjhoa** [comment]
@mixonic 
Sorry for the late reply
Test is ok on latest version of ember-table so this PR only add a test to cover this specific case.

**Comment by mixonic** [comment]
@ctjhoa Do you think this issue is only relevant to older Ember versions? I'm surprised this test would only fail for 2.8 and 2.12.

If you want to set up some time to pair or discuss please shoot me an email (address is on my profile) I'd happily set up the time.

**Comment by ctjhoa** [comment]
@mixonic
Older Ember versions timeout when rendering a high number of rows.
Decreasing the number of generated rows fix the issue on older Ember versions.
Probably due to an old rendering engine.

Tests are successful except ember-beta & ember-canary which fail but caused by those changes.
```
------ RESULTS ------

Scenario ember-lts-2.8: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-compatibili… │ 1.2.1              │ 1.2.1                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ Not Installed      │ Not Installed                │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-angle-brack… │ Not Installed      │ Not Installed                │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ bower              │ ^1.8.2             │ 1.8.12                       │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember              │ components/ember#… │ 2.8.3+c4330341               │ bower    │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

DEPRECATION: The next major version of `ember-try` (v2.0) will drop support for installing dependencies with `bower`.
Scenario ember-lts-2.12: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~2.12.0            │ 2.12.2                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-2.18: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~2.18.0            │ 2.18.2                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.4: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.4.0             │ 3.4.8                        │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.8: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.8.0             │ 3.8.3                        │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.12: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.12.0            │ 3.12.4                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.16: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.16.0            │ 3.16.10                      │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.20: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.20.0            │ 3.20.7                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-lts-3.24: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ ~3.24.0            │ 3.24.5                       │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-release: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ https://s3.amazon… │ 3.28.0-release+f243a4f0      │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-beta: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ https://s3.amazon… │ 4.0.0-beta.4.beta+52c81a56   │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-canary: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-source       │ https://s3.amazon… │ 4.1.0-alpha.1.canary+2d1033… │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-default: SUCCESS
Command run: ember test
Scenario ember-production: SUCCESS
Command run: ember test -e production
Scenario ember-default-docs: SUCCESS
Command run: ember test --filter="Acceptance | docs"
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-data         │ ~3.24.0            │ 3.24.2                       │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-cli-addon-d… │ ^1.0.0             │ 1.0.0                        │ yarn     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-cli-addon-d… │ ^1.0.0             │ 1.0.0                        │ yarn     │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘


2 scenarios failed
13 scenarios succeeded
15 scenarios run
```

**Pull request merged**